lib/ical/timezone_service.js

```diff
@@ -69,24 +69,39 @@ const TimezoneService = {
   /**
    * Registers a timezone object or component.
    *
+   * @param {Component|Timezone} timezone
+   *        The initialized zone or vtimezone.
+   *
    * @param {String=} name
    *        The name of the timezone. Defaults to the component's TZID if not
    *        passed.
-   * @param {Component|Timezone} timezone
-   *        The initialized zone or vtimezone.
    */
-  register: function(name, timezone) {
+  register: function(timezone, name) {
     if (zones === null) {
       this.reset();
     }
 
-    if (name instanceof Component) {
-      if (name.name === 'vtimezone') {
-        timezone = new Timezone(name);
+    // This avoids a breaking change by the change of argument order
+    // TODO remove in v3
+    if (typeof timezone === "string" && name instanceof Timezone) {
+      [timezone, name] = [name, timezone];
+    }
+
+    if (!name) {
+      if (timezone instanceof Timezone) {
         name = timezone.tzid;
+      } else {
+        if (timezone.name === 'vtimezone') {
+          timezone = new Timezone(timezone);
+          name = timezone.tzid;
+        }
       }
     }
 
+    if (!timezone.tzid && !name) {
```

Comment:
```suggestion
    if (!name) {
```
You already check for the timezone id in the block above and we require the name below.